src/testRunner/unittests/tsserver/completions.ts

```diff
@@ -246,5 +246,136 @@ export interface BrowserRouterProps {
                 }
             });
         });
+
+        describe("test of src/services/completions.ts#getCompletionData (#37546)", () => {

```

Comment:
Instead of thi these need to be fourslash test that verifies the result since the change is to language service and not tsserver.

Reply:
thanks for reply :-)

OK.
But I need to study "fourslash test"

Reply:
Eg. https://github.com/microsoft/TypeScript/blob/master/tests/cases/fourslash/completionAtDottedNamespace.ts

Reply:
Thanks for the info :-)

I read some source code of fourslash test but writing code must be done at a later date

because i goto sleep 😪